### DISCUSSION TITLE
Running non-ssl protocols over ssl port (i.e. using SSH over Nginx using $ssl_preread_protocol )

### DISCUSSION THREAD

**Discussion by maltokyo** [nginx-proxy/nginx-proxy#2379] [Ideas]
This is more of a feature request (or in case it is possible, a how-to request - I tried to connect in the google group, but seems it is not active anymore).  

**This article describes exactly what I want to do:**
https://www.nginx.com/blog/running-non-ssl-protocols-over-ssl-port-nginx-1-15-2/

Basically, for one of my domain names (vhost), I would like port 443 to be accepted and redirect connections to my SSH server, rather than a http/https service.

The code to do it is in the article and pasted below (checks if the 443 packets are destined for https or SSH and redirects accordingly, but I am unsure where to insert this or how to use it with nginx-proxy).  Could someone please give me a hint, what I would need to do to get this working for just one domain name?

```
stream {
    upstream ssh {
        server 192.0.2.1:22;
    }

    upstream web {
        server 192.0.2.2:443;
    }

    map $ssl_preread_protocol $upstream {
        default ssh;
        "TLSv1.2" web;
    }

    # SSH and SSL on the same port
    server {
        listen 443;

        proxy_pass $upstream;
        ssl_preread on;
    }
}
```
Any advice or help would be really appreciated.

**Comment by BBUBBA** [comment]
Modify File.

/etc/nginx/nginx.confg
